src/mscorlib/mscorlib.shared.sources.props

```diff
@@ -753,6 +755,29 @@
     <ThreadingSources Include="$(BclSourcesRoot)\System\Threading\Tasks\TaskToApm.cs" />
     <ThreadingSources Condition="'$(FeatureCominterop)' == 'true'" Include="$(BclSourcesRoot)\System\Threading\Tasks\IAsyncCausalityTracerStatics.cs" />
   </ItemGroup>
+  <ItemGroup Condition="'$(FeatureCoreFxFileStream)' != 'true'">
+    <FileStreamSources Include="$(BclSourcesRoot)\System\IO\FileStream.cs" />
+    <SafehandleSources Include="$(BclSourcesRoot)\Microsoft\Win32\SafeHandles\SafeFileHandle.cs" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(FeatureCoreFxFileStream)' == 'true'">
+    <FileStreamSources Include="$(CoreFxSourcesRoot)\System\IO\FileStream.cs" />
+    <FileStreamSources Include="$(CoreFxSourcesRoot)\System\IO\FileStream.NetStandard17.cs" />
+    <FileStreamSources Include="$(CoreFxSourcesRoot)\System\IO\Error.cs" />
+    <FileStreamSources Include="$(CoreFxSourcesRoot)\System\IO\StreamHelpers.ArrayPoolCopy.cs" />
+    <SafehandleSources Condition="'$(TargetsUnix)' == 'true'" Include="$(CoreFxSourcesRoot)\Microsoft\Win32\SafeHandles\SafeFileHandle.Unix.cs" />
+    <SafehandleSources Condition="'$(TargetsUnix)' != 'true'" Include="$(CoreFxSourcesRoot)\Microsoft\Win32\SafeHandles\SafeFileHandle.Windows.cs" />
+    <FileStreamSources Condition="'$(TargetsUnix)' == 'true'" Include="$(CoreFxSourcesRoot)\System\IO\FileStream.Unix.cs" />
+    <FileStreamSources Condition="'$(TargetsUnix)' != 'true'" Include="$(CoreFxSourcesRoot)\System\IO\FileStream.Win32.cs" />
+    <FileStreamSources Condition="'$(TargetsUnix)' != 'true'" Include="$(CoreFxSourcesRoot)\System\IO\FileStreamCompletionSource.Win32.cs" />
+    <FileStreamSources Condition="'$(TargetsUnix)' != 'true'" Include="$(CoreFxSourcesRoot)\System\IO\Win32Marshal.cs" />
+    <ThreadingSources Condition="'$(TargetsUnix)' != 'true'" Include="$(CoreFxSourcesRoot)\Microsoft\Win32\SafeHandles\SafeThreadPoolIOHandle.cs" />
+    <ThreadingSources Condition="'$(TargetsUnix)' != 'true'" Include="$(CoreFxSourcesRoot)\System\Threading\DeferredDisposableLifetime.cs" />
+    <ThreadingSources Condition="'$(TargetsUnix)' != 'true'" Include="$(CoreFxSourcesRoot)\System\Threading\Win32ThreadPoolBoundHandle.cs" />
+    <ThreadingSources Condition="'$(TargetsUnix)' != 'true'" Include="$(CoreFxSourcesRoot)\System\Threading\Win32ThreadPoolNativeOverlapped.cs" />
```

Comment:
CoreCLR has own thread pool for historic reason. The Win32ThreadPool should not be used for CoreCLR.  The corefx always uses the ClrThreadPoolBoundHandle for CoreCLR:
https://github.com/dotnet/corefx/blob/b137b42f82ec531dfeb386eb2008a41ae44f22ed/src/System.Threading.Overlapped/src/System.Threading.Overlapped.csproj#L39

The Win32ThreadPool will need to be brought to CoreRT though.